src/Nethermind/Nethermind.Consensus.AuRa/Validators/ContractBasedValidator.cs

```diff
@@ -129,6 +129,15 @@ public override void OnBlockProcessingStart(Block block, ProcessingOptions optio
             }
             else
             {
+                if (mainChainProcessing && skippingBlocks)
+                {
+                    bool loadedValidatorsAreSameInStore = (ValidatorStore.GetValidators()?.SequenceEqual(Validators) == true);
+                    if (!loadedValidatorsAreSameInStore)
+                    {
+                        ValidatorStore.SetValidators(block.Number - 1, Validators);
```

Comment:
will block.Number -1 always work or sometimes you would need to be recursive?

Reply:
I think not... but moved it to finalized block just to be sure.